rte/mo_rte_lw.F90

```diff
@@ -178,6 +182,17 @@ function rte_lw(optical_props, top_at_1, &
       return
     end if
 
+    !
+    ! Check extents and values of transport angle secants if supplied
+    !
+    if (present(lw_Ds)) then
```

Comment:
You will want to ensure that `lw_Ds` are supplied only if `optical_props` is of type `_1scl`.

Reply:
Robert - can you help me with the logic we should have in this part of the code? We now have use_2stream and lw_Ds as arguments to rte_lw.  But they aren't used at the same time....

Reply:
okay - I will add these checks.

Reply:
It should be an error if `use_2stream` is supplied but the type of optical_props is not `_2str` is of course what I meant

Reply:
I knew that code I had in there was no good but I do want to test before hand - what if we add more scenarios to rte_lw? For example, if I'm in the 1scl code block -> then I have to check that "use_2stream" is false and any other optional variables that we might add later on rather than check that is use_2stream is positive, the class must be 2str.  Well, I'll change it because that coding was bad but illustrated the point - hence my request for you to check it.